src/topology/algebra/ordered.lean

```diff
@@ -1728,6 +1752,72 @@ begin
   exact tendsto_pow_neg_at_top (by exact_mod_cast this)
 end
 
+lemma tendsto_const_mul_zero_of_tendsto_zero {f : α → α} {c : α}
+  (h : tendsto f at_top (𝓝 0)) : tendsto (λ x, c * f x) at_top (𝓝 0) :=
```

Comment:
Isn't true that any `linear_ordered_field` with an ordered topology is a topological field? Proving this once and for all would give access to many more lemmas of this kind.

Reply:
I couldn't find a definition of a topological field in mathlib (and I'm not sure this PR is the right place to add one), so for now I've just added a proof of `has_continuous_mul` for a `linear_ordered_field` with `order_topology` in a new section in `ordered.lean`. 

The proof (and some of the lemmas used to prove it) ended up being fairly lengthy, which I think may be due to heavy reliance on `mem_nhds_iff_exists_Ioo_subset` (and my limited knowledge of topology in general). If there is a better way to approach this let me know and I can try to refactor the proofs.

Reply:
The trick you are missing is using filter basis technology. First I suggest you make `a` explicit in `nhds_basis_Ioo`.
```lean
lemma nhds_basis_Ioo [no_top_order α] [no_bot_order α] (a : α) :
  (𝓝 a).has_basis (λ b : α × α, a ∈ Ioo b.1 b.2) (λ b, Ioo b.1 b.2) :=
nhds_basis_Ioo' (no_bot a) (no_top a)
```
This lemma is the best you can do without an algebraic structure on `α`. And it is the version of your `mem_nhds_iff_exists_pos_Ioo_subset` that is convenient to use (more about that below) so you can delete your version.

But, in the proofs we are talking about, `α` is a totally ordered field, so everything works as in the case of real. So you can add instead of your version, right before the continuous_mul section:
```lean
lemma nhds_basis_Ioo_pos {a : α} : (𝓝 a).has_basis (λ ε : α, (0 : α) < ε) (λ ε, Ioo (a-ε) (a+ε)) :=
⟨begin
  intros t,
  rw (nhds_basis_Ioo a).mem_iff,
  split,
  { rintros ⟨⟨l, u⟩, ⟨hl : l < a, hu : a < u⟩, h' : Ioo l u ⊆ t⟩,
    refine ⟨min (a-l) (u-a), _, _⟩,
    { apply lt_min ; linarith },
    { rintros x ⟨hx, hx'⟩,
      apply h',
      rw [sub_lt, lt_min_iff] at hx,
      rw [← sub_lt_iff_lt_add', lt_min_iff] at hx',
      split ; linarith } },
  { rintros ⟨ε, ε_pos, h⟩,
    exact ⟨(a-ε, a+ε), by simp [ε_pos], h⟩ },
end⟩

lemma nhds_basis_abs_sub_lt (a : α) : 
  (𝓝 a).has_basis (λ ε : α, (0 : α) < ε) (λ ε, {b | abs (b - a) < ε}) :=
begin
  convert nhds_basis_Ioo_pos,
  { ext ε,
    change abs (x - a) < ε ↔ a - ε < x ∧ x < a + ε,
    rw abs_lt,
    split ; intro h ; split ; linarith },
  assumption
end

variables (α)

lemma nhds_basis_zero_abs_sub_lt : 
  (𝓝 (0 : α)).has_basis (λ ε : α, (0 : α) < ε) (λ ε, {b | abs b < ε}) :=
by simpa using nhds_basis_abs_sub_lt (0 : α)

variables {α}
```
Now let's see filter basis magic in action. In order to check a limit claim, you only need to care about elements of the basis, both on source and target since we have both at hand.
```lean
lemma mul_tendsto_nhds_zero_right {x : α}  :
  tendsto (uncurry ((*) : α → α → α)) (𝓝 0 ×ᶠ 𝓝 x) $ 𝓝 0 :=
begin
  rw ((nhds_basis_zero_abs_sub_lt α).prod $ nhds_basis_abs_sub_lt x).tendsto_iff 
     (nhds_basis_zero_abs_sub_lt α),
  intros ε ε_pos,
  refine ⟨(ε/(2*(1 + abs x)), 1), ⟨_, zero_lt_one⟩, _⟩,
  { change 0 < ε / (2 * (1 + abs x)),
    sorry },
  { suffices : ∀ (a b : α), abs a < ε / (2 * (1 + abs x)) → abs (b - x) < 1 → abs (a * b) < ε,
    by simpa,
    sorry }
end
```
I didn't fill those sorries, but I believe they are true for any totally ordered field, although you may get less help from linarith then what you would get with real numbers. Hopefully this should help you simplifying a lot of proofs.

Reply:
Thanks for this helpful explanation, this does seem to have cleaned up a lot of the proofs, and makes me think filters are a more powerful abstraction than I had thought they were.